### DISCUSSION TITLE
NPM Update from 9.8.0 to 9.10.0 Fails at path /var/directus/node_modules/sharp

### DISCUSSION THREAD

**Discussion by jabstein** [directus/directus#13167] [Archive]
### Describe the Bug

Trying to update directus to latest build, but keep running into errors.

The directus is deployed via digital ocean one click and has been updated from 9.4 to 9.8 before, but gives errors on 9.8 to 9.10

### To Reproduce

login to server directus directory and initiate npm update

### Errors Shown

root@directusServer:/var/directus# npm update
npm WARN deprecated @types/pino@7.0.5: This is a stub types definition. pino provides its own type definitions, so you do not need this installed.
npm WARN deprecated @types/cssnano@5.1.0: This is a stub types definition. cssnano provides its own type definitions, so you do not need this installed.
npm WARN deprecated fastify-warning@0.2.0: This module renamed to process-warning
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm ERR! code 1
npm ERR! path /var/directus/node_modules/sharp
npm ERR! command failed
npm ERR! command sh -c (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
npm ERR! sharp: Are you trying to install as a root or sudo user? Try again with the --unsafe-perm flag
npm ERR! sharp: Please see https://sharp.pixelplumbing.com/install for required dependencies
npm ERR! sharp: Installation error: EACCES: permission denied, mkdir '/root/.npm'

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-05-06T16_02_31_316Z-debug-0.log

<details>
<summary>More info</summary>

### What version of Directus are you using?

9.8.0

### What version of Node.js are you using?

14.15.4

### What database are you using?

postgresql

### What browser are you using?

chrome

### How are you deploying Directus?

digital ocean server one click

</details>

**Comment by azrikahar** [comment]
This seems to be related with the Sharp package. Can you try the resolution steps offered by them based on the link present in your error log? Specifically this recommendation:

> When using npm v6 or earlier, the `npm install --unsafe-perm` flag must be used when installing as `root` or a `sudo` user.

because you are on Node `14.15.4` and you should be on npm version `6.14.10` so hopefully the recommendation applies to your scenario: 

![chrome_9wtyHhR5lI](https://user-images.githubusercontent.com/42867097/167260462-55de0b1e-2d61-415e-98b1-07a1c3c06cdd.png)

([link to the above table](https://nodejs.org/en/download/releases/))

**Reply by jabstein** [reply]
I downgraded npm to the 6 ver and it works as it should now. Thank you.
